packages/blocks/src/store/process-block-type.js

```diff
@@ -33,6 +33,37 @@ const LEGACY_CATEGORY_MAPPING = {
 	layout: 'design',
 };
 
+/**
+ * Merge block variations bootstrapped from the server and client.
+ *
+ * When a variation is registered in both places, its properties are merged.
+ *
+ * @param {Array} bootstrappedVariations - A block type variations from the server.
+ * @param {Array} clientVariations       - A block type variations from the client.
+ * @return {Array} The merged array of block variations.
+ */
+function mergeBlockVariations(
+	bootstrappedVariations = [],
+	clientVariations = []
+) {
+	const result = [ ...bootstrappedVariations ];
+
+	clientVariations.forEach( ( clientVariation ) => {
+		const index = result.findIndex(
+			( bootstrappedVariation ) =>
+				bootstrappedVariation.name === clientVariation.name
+		);
+
+		if ( index !== -1 ) {
+			result[ index ] = { ...result[ index ], ...clientVariation };
```

Comment:
Do we intentionally want to keep the bootstrapped variation data and mingle it with the client side one? What if on the client we intentionally want to keep a variation property empty, but the server declares it?

Reply:
> The server information should not be ever wrong, the client's job is to improve it, not overwrite.

This is what I wanted to confirm. If that's the intention, then we're good here.